lfs/hook.go

```diff
@@ -27,17 +28,30 @@ func (h *Hook) Exists() bool {
 }
 
 // Path returns the desired (or actual, if installed) location where this hook
-// should be installed, relative to the local Git directory.
+// should be installed. It returns an absolute path in all cases.
 func (h *Hook) Path() string {
-	return filepath.Join(config.LocalGitDir, "hooks", string(h.Type))
+	return filepath.Join(h.Dir(), h.Type)
+}
+
+// Dir returns the directory used by LFS for storing Git hooks. By default, it
+// will return the hooks/ sub-directory of the local repository's .git
+// directory. If `core.hooksPath` is configured and supported (Git verison is
+// greater than "2.9.0"), it will return that instead.
+func (h *Hook) Dir() string {
+	customHooksSupported := git.Config.IsGitVersionAtLeast("2.9.0")
+	if hp, ok := config.Config.GitConfig("core.hooksPath"); ok && customHooksSupported {
+		return hp
+	}
```

Comment:
This is pretty nitpicky, but there's no point in even checking the git config if `customHooksSupported == false`. Probably fine to keep as-is from a code readability standpoint.